src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```
@@ -9,6 +10,19 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("User with this login is already registered");
+        }
+        if (user.getLogin().length() <= 6) {
+            throw new RegistrationException("Login must contain at least 6 characters");
+        }
+        if (user.getPassword().length() <= 6) {
+            throw new RegistrationException("Password must contain at least 6 characters");
+        }
+        if (user.getAge() < 18) {
```

This line checks age with `if (user.getAge() < 18)` but does not check for null age and will throw NullPointerException for null ages. The checklist requires handling nulls explicitly and forbids using NPEs for validation failures: "Don't use `NullPointerException`, because in such case we can't distinguish was this exception because of our mistake or because of validation." Also use a `MIN_AGE` constant and check `user.getAge() < MIN_AGE`. Please add a null check for age and throw RegistrationException with an informative message for null/invalid ages.